gitpod-io/openvscode-server
# unify ports view between web and desktop

Relate PR https://github.com/gitpod-io/gitpod/issues/7408

## TODO
- [x] ‼️  Bring preview port view back, and add a deprecation warning
- [x] ‼️ Observe, add another field `isWebview` in event `vscode_execute_command_gitpod_ports`
- [x] ‼️ Add svelte build to build command
- [x] ⁉️ Use `@vscode/webview-ui-toolkit`
- [ ] Doc tracking event data in notion
- [x] Bring fast public/private switch back
- [x] Clean up PR - remve some useless `TODO`
- [x] ~~Use beautiful svg icons🙈~~ maybe not necessary? 
- [x] Other `TODO(hw)` in code
- [x] Flip icons of public/private and tunnel/un-tunnel
- [x] Focus on new port view if user clicks port in status bar or DEPRECATE notice in Remote Explorer

## How to Test

https://hw-vs-debug-2.preview.gitpod-dev.com/workspaces

- Make sure use `latest` code
- Start a workspace
- Check if it will focus on new port view when we click DEPRECATE msg in remote explorer
- Check if click status bar ports info will focus on new port view
- Check if new ports view works as expected and data align to our previous port view

### Example cases for port features

Open browser `Remote Explorer`(previous) and new `Ports View`(new), to see if they change together

- Make port public and private
- Open browser and preview
- Copy in new port view

Open local companion with prev env host
- See if description of port changed
- See if tunnul related icons showup
   - (It should be, but it won't, don't know why, I tested with production env latest code not worked too (stable worked)
 - See if open preview and open browser turn into `localhost:xxxx`
